docs/tutorials/contour.md

```diff
@@ -0,0 +1,175 @@
+# Configuring ExternalDNS to use the Contour IngressRoute Source
+This tutorial describes how to configure ExternalDNS to use the Contour IngressRoute source.
+It is meant to supplement the other provider-specific setup tutorials.
+
+### Manifest (for clusters without RBAC enabled)
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: external-dns
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: external-dns
+    spec:
+      containers:
+      - name: external-dns
+        image: registry.opensource.zalan.do/teapot/external-dns:latest
+        args:
+        - --source=service
+        - --source=ingress
+        - --source=contour-ingressroute
+        - --contour-load-balancer=custom-contour-namespace/custom-contour-lb # load balancer service to be used. Omit to use the default (heptio-contour/contour)
```

Comment:
@jonasrmichel You mentioned that there's potential support for populating `status` on Ingress objects by `contour`. Would an `IngressRoute` have that field, too?

Comparing this to `nginx-ingress-controller`, then this configuration would be part of `contour` which would put the corresponding Service's LB in the `status` section of the `IngressRoute`.

Why do we need it in ExternalDNS?

Reply:
@linki -- Good observation, and great point. `IngressRoute` objects do in fact have a `status` field, but its schema ([source](https://github.com/heptio/contour/blob/165907e219b24bf8ce65cb6555317eaef337eed7/apis/contour/v1beta1/ingressroute.go#L124), [doc](https://github.com/heptio/contour/blob/5046e28f053d1ff88e174fe4593a89bcd0c2ce0e/docs/ingressroute.md#status-reporting)) is not equivalent to that of the `Ingress` type. The `IngressRoute` `status` field only includes two text fields that indicate an `IngressRoute`'s validity.

As far as I know, there are no plans to expand the `IngressRoute` `status` field. Until then, a dedicated ExternalDNS source implementation is required.

Thoughts?

Reply:
@linki -- Any follow up thoughts? Or may we resolve this conversation and close the PR?

Reply:
Thanks for the explanation. We can do it like you propose then.